api/kyverno/v1alpha1/cleanup_policy_types.go

```diff
@@ -72,6 +86,13 @@ type ClusterCleanupPolicy struct {
 	Status CleanupPolicyStatus `json:"status,omitempty"`
 }
 
+// Validate implements programmatic validation
+func (p *ClusterCleanupPolicy) Validate(clusterResources sets.String) (errs field.ErrorList) {
+	errs = append(errs, kyvernov1.ValidatePolicyName(field.NewPath("name"), p.Name)...)
```

Comment:
This should be `metadata.name` not just name.

Reply:
I think we can get the name directly; in this case, we are also getting the name with `p.Name`. Also, here similarly: https://github.com/kyverno/kyverno/blob/main/api/kyverno/v1/clusterpolicy_types.go#L115

Reply:
It's not about the way we access it in code, it's about the field path in the manifest and the path is `metadata.name`.